misc/random.c

```diff
@@ -42,33 +39,29 @@ static inline uint64_t splitmix64(uint64_t *const x)
     return z ^ (z >> 31);
 }
 
-void mp_rand_seed(uint64_t seed)
+struct mp_rand_state mp_rand_seed(uint64_t seed)
 {
+    struct mp_rand_state ret = {0};
 #ifdef FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION
     seed = 42;
 #endif
 
     if (seed == 0) {
-        uint8_t buf[sizeof(seed)];
-        if (av_random_bytes(buf, sizeof(buf)) < 0) {
-            seed = mp_raw_time_ns();
-        } else {
-            memcpy(&seed, buf, sizeof(seed));
-        }
+        seed = mp_raw_time_ns();
+        seed ^= (uint64_t)&mp_rand_seed; // ASLR, hopefully
+        seed += (uint64_t)&ret; // stack pos
```

Comment:
unless a performance problem with calling `av_random_bytes` **once** can be demonstrated I am against such hacks

Reply:
Added av_random_bytes back which initializes the whole state now https://github.com/mpv-player/mpv/pull/16164/commits/68ac9ccbfe13a962e9b2a120c9cde358dede1d10.

The "hacks" are now only in the fallback case. I've tested the fallback by disabling both the av_random_bytes() and mp_raw_time_ns() call and (unsurprisingly) they produced very different shuffles on my system (and most likely on most other systems too).

At the very least, this won't *worsen* the fallback case, which didn't fix what c365e2f7b17e6a890827802e063b823bf6e4385f was trying to do. And will improve the situation with very high likelihood.